ciao-controller/api/api.go

```diff
@@ -408,7 +409,12 @@ func mapExternalIP(c *Context, w http.ResponseWriter, r *http.Request) (Response
 		return errorResponse(err), err
 	}
 
-	err = c.MapAddress(req.PoolName, req.InstanceID)
+	tenantID, ok := vars["tenant"]
+	if !ok {
+		tenantID = "public"
```

Comment:
I'd rather not overload tenantID == "public" to mean admin for the MapAddress() case. I'd rather go with "" or something like "admin". But see my other comment below...